ts/Series/GeoHeatmap/GeoHeatmapSeries.ts

```diff
@@ -355,27 +359,58 @@ class GeoHeatmapSeries extends MapSeries {
                     // Always calculate new canvas for Orthographic projection
                     mapView.projection.options.name === 'Orthographic'
                 ) {
-                    series.isDirtyCanvas = true;
                     const canvasWidth = canvas.width = ~~(360 / colsize) + 1,
                         canvasHeight = canvas.height = ~~(180 / rowsize) + 1,
                         canvasArea = canvasWidth * canvasHeight,
-                        pixelData = new Uint8ClampedArray(canvasArea * 4);
-                    series.directTouch = false; // Needed for tooltip
+                        pixelData = new Uint8ClampedArray(canvasArea * 4),
+                        // Guess if we have to round lon/lat with this data
+                        { lat = 0, lon = 0 } = points[0].options,
+                        unEvenLon = lon % rowsize !== 0,
+                        unEvenLat = lat % colsize !== 0,
+                        getAdjustedLon = (
+                            unEvenLon ?
+                                (lon: number): number => (
+                                    Math.round(lon / rowsize) * rowsize
+                                ) :
+                                (lon: number): number => lon
+                        ),
+                        getAdjustedLat = (
+                            unEvenLat ?
+                                (lat: number): number => (
+                                    Math.round(lat / colsize) * colsize
+                                ) :
+                                (lat: number): number => lat
+                        ),
+                        pointsLen = points.length;
+
+                    if (unEvenLon || unEvenLat) {
+                        error(
```

Comment:
I think maybe we should add a `Highcharts warning:` (or similar) and possibly also the values here. Looks like `error` has a params parameter that prints like this:

<img width="588" alt="image" src="https://github.com/user-attachments/assets/c078cf59-7e70-40c9-8a79-9a2a78e0b975" />